### PR TITLE
update formatter to handle record closing braces correctly

### PR DESCRIPTION
**Pull request by homberghp**
#7043 is still open, due to lacking tests.
Here is the test.

### CODE REVIEW

**Comment by homberghp** [comment]
Added solution to the problem.

This solves #7043

File: java/java.source/src/org/netbeans/modules/java/ui/FmtSpaces.java [hunk 1]
```diff
@@ -336,10 +336,70 @@ public String toString() {
     }
     
     private static final class SpacesCategorySupport extends CategorySupport {
-
+        // copied from Bundle.properties for easier maintanance
+        static String sampleSpaces ="""
```
Comment (mbien) [inline java/java.source/src/org/netbeans/modules/java/ui/FmtSpaces.java:340]: we shouldn't inline message bundle values, I suppose this was only done for debugging purposes? Changes to this file should be dropped IMO.

File: java/java.source.base/test/unit/src/org/netbeans/modules/java/source/save/RecordFormattingTest.java [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+        EditorCookie ec = (EditorCookie)testSourceDO.getCookie(EditorCookie.class);
+        Preferences preferences = MimeLookup.getLookup(JavaTokenId.language().mimeType()).lookup(Preferences.class);
+        preferences.putBoolean("spaceWithinMethodDeclParens", spacesInMethodDecl);
+        preferences.putInt("blankLinesAfterClassHeader", 0);
+
+        final Document doc = ec.openDocument();
+        doc.putProperty(Language.class, JavaTokenId.language());
+//        System.err.println("\n============== start test ========================");
+        System.err.println("golden length = " + golden.trim().length());
+        System.err.println("content length = " + content.trim().length());
```
Comment (mbien) [inline java/java.source.base/test/unit/src/org/netbeans/modules/java/source/save/RecordFormattingTest.java:135]: no println in tests please. We have hours of tests and have to keep the output manageable

**Review by mbien — COMMENTED** [review]
hi @homberghp, first of all: thanks for working on this and contributing the fix for the bug you found.

If I see this correctly, the changeset (outside of the tests) are about 2 lines, rest are whitespace changes. We usually try to avoid making unrelated changes when fixing bugs. 

Please change the changes to only those lines which matter for the fix - feel free to squash and force push into this PR.

**Comment by homberghp** [comment]
@mbien: thank you for the time spent reviewing my pull request.

I will close this pull request and create a new one, based on your suggestions.
That is easier than applying all kinds of git magic.

**Pull request closed without merging**